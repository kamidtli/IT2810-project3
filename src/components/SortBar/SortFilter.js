import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    flexWrap: 'wrap',
  },
  formControl: {
    margin: theme.spacing(1),
    minWidth: 120,
  },
  selectEmpty: {
    marginTop: theme.spacing(2),
  },
}));

export default function SimpleSelect(props) {
  const classes = useStyles();
  const [values, setValues] = React.useState({
    sortValue: '',
  });

  const handleChange = (event) => {
    let newValue = event.target.value;
    if (newValue === '') {
      newValue = '-released';
    }
    setValues((oldValues) => ({
      ...oldValues,
      [event.target.name]: event.target.value,
    }));
    props.onUpdate(newValue);
  };

  return (
    <form className={classes.root} autoComplete="off">
      <FormControl className={classes.formControl}>
        <Select
          value={values.sortValue}
          onChange={handleChange}
          name="sortValue"
          displayEmpty
          className={classes.selectEmpty}
        >
          <MenuItem value="">Release date</MenuItem>
          <MenuItem value="title">Alphabetic</MenuItem>
          <MenuItem value="imdb">Rating</MenuItem>
        </Select>
      </FormControl>
    </form>
  );
}
