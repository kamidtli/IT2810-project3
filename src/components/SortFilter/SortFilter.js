import React from 'react';
import { makeStyles } from '@material-ui/styles';
import { MenuItem, FormControl, Select } from '@material-ui/core/';
import { connect } from 'react-redux';

const useStyles = makeStyles((theme) => ({
  formControl: {
    margin: theme.spacing(1),
    minWidth: 120,
    width: 150,
  },
}));

function SimpleSelect(props) {
  const classes = useStyles();
  const [values, setValues] = React.useState({
    sortValue: props.sortValue,
  });

  const handleChange = (event) => {
    let newValue = event.target.value;
    if (newValue === '') {
      newValue = '-imdb';
    }
    // Set values in state
    setValues((oldValues) => ({
      ...oldValues,
      [event.target.name]: event.target.value,
    }));
    props.updateSortFilter(newValue); // Update sort value in redux store
    props.onUpdate(newValue); // Runs the callback function to update sort value
  };

  return (
    <form className={classes.root} autoComplete="off">
      <FormControl className={classes.formControl}>
        <Select
          value={values.sortValue}
          onChange={handleChange}
          name="sortValue"
          displayEmpty
        >
          <MenuItem value="">Rating (Highest first)</MenuItem>
          <MenuItem value="imdb">Rating (Lowest first)</MenuItem>
          <MenuItem value="-released">Released (Newest first)</MenuItem>
          <MenuItem value="released">Released (Oldest first)</MenuItem>
          <MenuItem value="title">Alphabetic</MenuItem>
        </Select>
      </FormControl>
    </form>
  );
}

const mapStateToProps = (state) => ({
  sortValue: state.sortValue,
});

const mapDispatchToProps = (dispatch) => ({
  updateSortFilter: (sortValue) => dispatch({ type: 'NEW_SORT_VALUE', sortValue }),
});

export default connect(mapStateToProps, mapDispatchToProps)(SimpleSelect);
