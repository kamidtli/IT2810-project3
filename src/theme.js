import { createMuiTheme, responsiveFontSizes } from '@material-ui/core/styles';

const theme = createMuiTheme({
  palette: {
    primary: {
      main: '#F6AE2D',
    },
    secondary: {
      main: '#000d34',
    },
  },
  status: {
    danger: 'orange',
  },
});

const responsiveTheme = responsiveFontSizes(theme);

export default responsiveTheme;
