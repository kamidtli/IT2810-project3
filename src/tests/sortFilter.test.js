import React from 'react';
import { shallow } from 'enzyme';
import { ThemeProvider } from '@material-ui/styles';
import SortFilter from '../components/SortFilter/SortFilter';
import theme from '../theme';

describe('Sort Filter component', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<ThemeProvider theme={theme}><SortFilter /></ThemeProvider>);
    expect(wrapper.toJSON).toMatchSnapshot();
  });
});
