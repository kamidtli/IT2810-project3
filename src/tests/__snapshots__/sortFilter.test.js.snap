// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort Filter component renders without crashing 1`] = `undefined`;
